recipes/ruby/all/conanfile.py

```diff
@@ -122,75 +178,97 @@ def generate(self):
 
         tc.generate()
 
+        if is_msvc(self):
+            vc = VCVars(self)
+            vc.generate()
+
     def build(self):
         apply_conandata_patches(self)
 
-        at = Autotools(self)
+        autotools = Autotools(self)
 
-        build_script_folder = self._source_subfolder
+        build_script_folder = self.source_folder
         if is_msvc(self):
             self.conf["tools.gnu:make_program"] = "nmake"
             build_script_folder = os.path.join(build_script_folder, "win32")
 
             if "TMP" in os.environ:  # workaround for TMP in CCI containing both forward and back slashes
                 os.environ["TMP"] = os.environ["TMP"].replace("/", "\\")
 
-        with tools.vcvars(self):
-            at.configure(build_script_folder=build_script_folder)
-            at.make()
+        autotools.configure(build_script_folder=build_script_folder)
+        autotools.make()
 
     def package(self):
         for file in ["COPYING", "BSDL"]:
-            self.copy(file, dst="licenses", src=self._source_subfolder)
+            copy(self, pattern=file, src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
 
-        at = Autotools(self)
-        with tools.vcvars(self):
-            if cross_building(self):
-                at.make(target="install-local")
-                at.make(target="install-arch")
-            else:
-                at.install()
+        autotools = Autotools(self)
+        if cross_building(self):
+            autotools.make(target="install-local")
+            autotools.make(target="install-arch")
+        else:
+            autotools.install()
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, pattern="*.pdb", folder=os.path.join(self.package_folder, "lib"))
 
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.pdb")
+        # install the enc/*.a / ext/*.a libraries
+        if not self.options.shared and self.options.with_static_linked_ext:
+            for dirname in ['ext', 'enc']:
+                dst = os.path.join('lib', dirname)
+                copy(self, '*.a', src=dirname, dst=os.path.join(self.package_folder, dst), keep_path=True)
+                copy(self, '*.lib', src=dirname, dst=os.path.join(self.package_folder, dst), keep_path=True)
 
     def package_info(self):
         binpath = os.path.join(self.package_folder, "bin")
         self.output.info(f"Adding to PATH: {binpath}")
```

Comment:
```suggestion
```
These logging statements are no longer recommended as they get logged even every time the package is used transitively.
Also, move `self.env_info.PATH.append(binpath)` under the `# TODO: to remove in conan v2` block, as it's no longer required in Conan v2.